api/src/main/java/com/datastrato/gravitino/rel/expressions/distributions/Distributions.java

```diff
@@ -26,6 +26,15 @@ public static Distribution even(int number, Expression... expressions) {
     return new DistributionImpl(Strategy.EVEN, number, expressions);
   }
 
+  /**
+   * Create a distribution by hashing the data.
+   *
+   * @return The created hash distribution
+   */
+  public static Distribution hash(Expression... expressions) {
```

Comment:
Why do we need to expose this method?

Reply:
The number is `0`. I'm confused about this.

Reply:
I prefer using `-1` to represent inifinite number.

Reply:
`inifinite number` is not a good choice I think, maybe `1` is optimal.

Reply:
0 or 1 means that we don't use partition or bucket. So I don't like to use them.